### PR TITLE
modules: hal_silabs: Add support for EFR32BG22C222F352GM40 SoC

### PR DESCRIPTION
**Pull request by wkhadgar**

### CODE REVIEW

File: gecko/CMakeLists.txt [hunk 1]
```diff
@@ -161,4 +165,4 @@ if(CONFIG_SOC_GECKO_USE_RAIL)
 
   # prebuilt libs
   add_prebuilt_library(librail      gecko/platform/radio/rail_lib/autogen/librail_release/librail_efr32xg${GECKO_SERIES_NUMBER}_gcc_release.a)
-endif()
+endif()
```
Comment (fkokosinski) [inline gecko/CMakeLists.txt:168]: Please remove this change.

File: gecko/CMakeLists.txt [hunk 2]
```diff
@@ -16,6 +16,10 @@ string(SUBSTRING ${CONFIG_SOC_SERIES} 7 2 GECKO_SERIES_NUMBER)
 
 set(SILABS_GECKO_PART_NUMBER ${CONFIG_SOC_PART_NUMBER})
 
+if (${CONFIG_BOARD} STREQUAL "bgm220pc_22wga2")
```
Comment (fkokosinski) [inline gecko/CMakeLists.txt:19]: This board doesn't seem to be defined anywhere. Could you create a PR to Zephyr that adds it?

**Review by fkokosinski — CHANGES_REQUESTED** [review]

**Review by wkhadgar — COMMENTED** [review]
Removed the end of file change

**Comment by tgorochowik** [comment]
@wkhadgar all changes in the HAL module require a corresponding PR in the main Zephyr repo that uses this code. So without support for this SoC in the main Zephyr repo we cannot merge this PR

**Comment by jerome-pouiller** [comment]
Closing because of inactivity.

**Pull request closed without merging**